ts/navigation/AuthenticationNavigator.tsx

```diff
@@ -0,0 +1,100 @@
+import CardSelectionScreen from "../screens/authentication/CardSelectionScreen";
+import CieAuthorizeDataUsageScreen from "../screens/authentication/cie/CieAuthorizeDataUsageScreen";
+import CieCardReaderScreen from "../screens/authentication/cie/CieCardReaderScreen";
+import CieConsentDataUsageScreen from "../screens/authentication/cie/CieConsentDataUsageScreen";
+import CieExpiredOrInvalidScreen from "../screens/authentication/cie/CieExpiredOrInvalidScreen";
+import CiePinLockedTemporarilyScreen from "../screens/authentication/cie/CiePinLockedTemporarilyScreen";
+import CiePinScreen from "../screens/authentication/cie/CiePinScreen";
+import CieWrongCiePinScreen from "../screens/authentication/cie/CieWrongCiePinScreen";
+import IdpLoginScreen from "../screens/authentication/IdpLoginScreen";
+import IdpSelectionScreen from "../screens/authentication/IdpSelectionScreen";
+import LandingScreen from "../screens/authentication/LandingScreen";
+import SpidCIEInformationScreen from "../screens/authentication/SpidCIEInformationScreen";
+import SpidInformationScreen from "../screens/authentication/SpidInformationScreen";
+import TestAuthenticationScreen from "../screens/authentication/TestAuthenticationScreen";
+import MarkdownScreen from "../screens/development/MarkdownScreen";
+import { AuthenticationParamsList } from "./params/AuthenticationParamsList";
+import ROUTES from "./routes";
+
+const Stack = createStackNavigator<AuthenticationParamsList>();
+
+const AuthenticationStackNavigator = () => (
+  <Stack.Navigator
+    initialRouteName={ROUTES.AUTHENTICATION_LANDING}
+    headerMode={"none"}
+    screenOptions={{ gestureEnabled: true }}
+  >
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_LANDING}
+      component={LandingScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_SELECTION}
+      component={IdpSelectionScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_CIE}
+      component={CardSelectionScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_LOGIN}
+      component={IdpLoginScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_IDP_TEST}
+      component={TestAuthenticationScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_SPID_INFORMATION}
+      component={SpidInformationScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.AUTHENTICATION_SPID_CIE_INFORMATION}
+      component={SpidCIEInformationScreen}
+    />
+
+    <Stack.Screen name={ROUTES.MARKDOWN} component={MarkdownScreen} />
+
+    <Stack.Screen
+      name={ROUTES.CIE_EXPIRED_SCREEN}
+      component={CieExpiredOrInvalidScreen}
+    />
+
+    <Stack.Screen name={ROUTES.CIE_PIN_SCREEN} component={CiePinScreen} />
+
+    <Stack.Screen
+      name={ROUTES.CIE_AUTHORIZE_USAGE_SCREEN}
+      component={CieAuthorizeDataUsageScreen}
+    />
+
+    <Stack.Screen
+      name={ROUTES.CIE_CARD_READER_SCREEN}
+      component={CieCardReaderScreen}
```

Comment:
Also in this screen we should remove `CompatNavigationProp` in order to use the right types. In this case the effect is a little more 💥 since without updating the screen, the CIE reading through nfc never starts and the user remains blocked in this screen, without the possibility to complete the login.

Reply:
After removed `CompatNavigationProp` the ts compiler produces this error that produce this error that I am investigating

```bash
ts/screens/authentication/cie/CieCardReaderScreen.tsx:512:16 - error TS2590: Expression produces a union type that is too complex to represent.

512 export default connect(mapStateToProps)(
                   ~~~~~~~~~~~~~~~~~~~~~~~~~
513   withConditionalView(
    ~~~~~~~~~~~~~~~~~~~~~~
... 
517   )
    ~~~
518 );
```